src/astro/databases/snowflake.py

```diff
@@ -57,6 +76,238 @@ def get_table_qualified_name(table: Table) -> str:  # skipcq: PYL-R0201
         qualified_name = ".".join(name for name in qualified_name_lists if name)
         return qualified_name
 
+    def check_optimised_path_and_transfer(
+        self,
+        source_file: File,
+        target_table: Table,
+        chunk_size: int = DEFAULT_CHUNK_SIZE,
+        if_exists: LoadExistStrategy = "replace",
+        **kwargs,
+    ) -> bool:
+        """
+        Checks if optimised path for transfer between File location to database exists
+        and if it does, it transfers it and returns true else false.
+        """
+        method_name = self.native_support.get(source_file.location.location_type)
+        if method_name:
+            transfer_method = self.__getattribute__(method_name)
```

Comment:
why use internal method

Reply:
This is trying to fetch the method based on file location type